### PR TITLE
nim: 0.18.0 -> 0.19.0

### PR DESCRIPTION
**Pull request by royneary**
- remove sqlite as dependency (I don't see why it's needed)
- run checkPhase after installPhase (at least one test assumes the
standard library in ../lib relative to the nim binary)
- the broken tests pass now or don't exist anymore
- two of the tests requiring network access pass now without network access
- the tests in manyloc do not download dependencies and are passing now

###### Motivation for this change
Update nim to 0.19.0.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by disassembler** [comment]
@GrahamcOfBorg build nim

**Comment by royneary** [comment]
I tried to fix the failure on Aarch64.
What can we do about the timeouts?
- On x86_64 the tests were interrupted after 1800 seconds, I don't think they were hanging.
- On darwin building nodejs took too long. I could try to disable nodejs for darwin, it's only needed for the tests that are not run anyway.

**Comment by Mic92** [comment]
If ofborg timeouts this is ok. Our hydra has longer timeouts.

**Comment by Mic92** [comment]
@GrahamcOfBorg build nim

File: pkgs/development/compilers/nim/default.nix [hunk 1]
```diff
@@ -30,10 +29,16 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     makeWrapper nodejs-slim-8_x tzdata coreutils
```
Comment (c0bw3b) [inline pkgs/development/compilers/nim/default.nix]: `makeWrapper tzdata coreutils` should be in `nativeBuildInputs` (they're needed only in installPhase and checkPhase)

**Comment by c0bw3b** [comment]
Build + tests successful on nixos-18.09 otherwise
[Full log here](https://gist.github.com/c0bw3b/798bb7b9ac79043bd957e79ed20457cd)

**Comment by royneary** [comment]
thanks for reviewing. I moved the packages you mentioned to `nativeBuildInputs` and `nodejs-slim`, too because it's only needed for the `checkPhase`. 

Also I updated `nodejs-slim-8_x` to `nodejs-slim-10_x`.

**Comment by gebner** [comment]
@royneary You need to add the `HOME=$TMPDIR` fix to the other nim packages as well.

@GrahamcOfBorg build nrpl

(`mosdepth` doesn't seem to compile with the new version.)

**Comment by royneary** [comment]
@gebner fixed.

**Comment by gebner** [comment]
`nrpl` builds for me now.

**Pull request merged**